include/xgboost/generic_parameters.h

```diff
@@ -12,6 +12,9 @@
 
 namespace xgboost {
 struct GenericParameter : public XGBoostParameter<GenericParameter> {
+  static int32_t constexpr kCpuId = -1;
```

Comment:
What is this used for?

Reply:
@RAMitchell Configuring GPU ID, originally in https://github.com/dmlc/xgboost/pull/5072 which I decided not to merge.  Besides it's better to replace repeated usage of `-1` with a named symbol.